pkg/prometheus/resource_selector.go

```diff
@@ -313,8 +313,8 @@ func validateRelabelConfig(p monitoringv1.PrometheusInterface, rc monitoringv1.R
 		if !(rc.Regex == "" || rc.Regex == relabel.DefaultRelabelConfig.Regex.String()) ||
 			!(rc.Modulus == uint64(0) ||
 				rc.Modulus == relabel.DefaultRelabelConfig.Modulus) ||
-			!(rc.Separator == "" ||
-				rc.Separator == relabel.DefaultRelabelConfig.Separator) ||
+			!(rc.Separator == nil ||
+				rc.Separator == &relabel.DefaultRelabelConfig.Separator) ||
```

Comment:
The addresses will always be different, I think you should dereference rc.Separator instead of getting the address of the default value.

Reply:
Thanks. I know what you mean. I was just too focused on the nil separator, since that's the breaking change. Will fix this in next commit.